File: ios/Classes/entity/GroupInfoEntity.swift
import ImSDK

//  Created by 蒋具宏 on 2020/2/12.
//  群信息实体
public class GroupInfoEntity : NSObject{
    
    /**
     * 类型
     */
    var groupType : String?;
    
    /**
     * 最大成员数量
     */
    var maxMemberNum : UInt32?;
    
    /**
     * 群主ID
     */
    var groupOwner : String?;
    
    /**
     * 自定义字段
     */
    var custom : [String : Data]?;
    
    /**
     * 群ID
     */
    var groupId : String?;
    
    /**
     * 返回码
     */
    var resultCode : Int32?;
    
    /**
     * 群通知
     */
    var groupNotification : String?;
    
    /**
     * 群成员数
     */
    var memberNum : UInt32?;
    
    /**
     * 返回结果
     */
    var resultInfo : String?;
    
    /**
     *  群头像
     */
    var faceUrl : String?;
    
    /**
     *  群名
     */
    var groupName : String?;
    
    /**
     * 加群选项
     */
    var addOption : GroupAddOptType?;
    
    /**
     * 创建时间
     */
    var createTime : UInt32?;
    
    /**
     * 群简介
     */
    var groupIntroduction : String?;
    
    /**
     * 是否禁言所有人
     */
    var silenceAll : Bool?;
    
    /**
     * 最后一次发消息时间
     */
    var lastMsgTime : UInt32?;
    
    /**
     * 在线成员数量
     */
    var onlineMemberNum : UInt32?;
    
    /**
     * 最后一次修改群资料时间
     */
    var lastInfoTime : UInt32?;
    
    override init() {
    }
    
    init(groupInfo : TIMGroupInfoResult) {
        super.init();
        self.groupOwner = groupInfo.owner;
        self.groupType = groupInfo.groupType;
        self.maxMemberNum = groupInfo.maxMemberNum;
        self.custom = groupInfo.customInfo;
        self.groupId = groupInfo.group;
        self.resultCode = groupInfo.resultCode;
        self.groupNotification = groupInfo.notification;
        self.memberNum = groupInfo.memberNum;
        self.resultInfo = groupInfo.resultInfo;
        self.faceUrl = groupInfo.faceURL;
        self.groupName = groupInfo.groupName;
        self.addOption = GroupAddOptType.getByTIMGroupAddOpt(type: groupInfo.addOpt);
        self.createTime = groupInfo.createTime;
        self.groupIntroduction = groupInfo.introduction;
        self.silenceAll = groupInfo.allShutup;
        self.lastMsgTime = groupInfo.lastMsgTime;
        self.onlineMemberNum = groupInfo.onlineMemberNum;
        self.lastInfoTime = groupInfo.lastInfoTime;
    }
}

File: android/src/main/java/top/huic/tencent_im_plugin/entity/GroupInfoEntity.java
package top.huic.tencent_im_plugin.entity;

import com.tencent.imsdk.TIMGroupAddOpt;
import com.tencent.imsdk.TIMMessage;
import com.tencent.imsdk.ext.group.TIMGroupDetailInfo;
import com.tencent.imsdk.ext.group.TIMGroupDetailInfoResult;

import java.io.Serializable;
import java.util.HashMap;
import java.util.Map;

public class GroupInfoEntity implements Serializable {
    private long unreadMessageNum;
    private static final String tag = "GroupInfoEntity";
    private String groupId = "";
    private String groupName = "";
    private String groupOwner = "";
    private String groupNotice = "";
    private String groupIntroduction = "";
    private String groupFaceUrl = "";
    private String groupType = "";
    private long createTime;
    private long lastInfoTime;
    private long lastMsgTime;
    private long memberNum;
    private long maxMemberNum;
    private long onlineMemberNum;
    private TIMGroupAddOpt addOption;
    private int intAddOption;
    private boolean isSilenceAll = false;
    private TIMMessage lastMsg = null;
    private Map<String, byte[]> custom = new HashMap();
    public static String getTag() {
        return tag;
    }

    public String getGroupId() {
        return groupId;
    }

    public void setGroupId(String groupId) {
        this.groupId = groupId;
    }

    public String getGroupName() {
        return groupName;
    }

    public void setGroupName(String groupName) {
        this.groupName = groupName;
    }

    public String getGroupOwner() {
        return groupOwner;
    }

    public void setGroupOwner(String groupOwner) {
        this.groupOwner = groupOwner;
    }

    public String getGroupNotice() {
        return groupNotice;
    }

    public void setGroupNotice(String groupNotice) {
        this.groupNotice = groupNotice;
    }

    public String getGroupIntroduction() {
        return groupIntroduction;
    }

    public void setGroupIntroduction(String groupIntroduction) {
        this.groupIntroduction = groupIntroduction;
    }

    public String getGroupFaceUrl() {
        return groupFaceUrl;
    }

    public void setGroupFaceUrl(String groupFaceUrl) {
        this.groupFaceUrl = groupFaceUrl;
    }

    public String getGroupType() {
        return groupType;
    }

    public void setGroupType(String groupType) {
        this.groupType = groupType;
    }

    public long getCreateTime() {
        return createTime;
    }

    public void setCreateTime(long createTime) {
        this.createTime = createTime;
    }

    public long getLastInfoTime() {
        return lastInfoTime;
    }

    public void setLastInfoTime(long lastInfoTime) {
        this.lastInfoTime = lastInfoTime;
    }

    public long getLastMsgTime() {
        return lastMsgTime;
    }

    public void setLastMsgTime(long lastMsgTime) {
        this.lastMsgTime = lastMsgTime;
    }

    public long getMemberNum() {
        return memberNum;
    }

    public void setMemberNum(long memberNum) {
        this.memberNum = memberNum;
    }

    public long getMaxMemberNum() {
        return maxMemberNum;
    }

    public void setMaxMemberNum(long maxMemberNum) {
        this.maxMemberNum = maxMemberNum;
    }

    public long getOnlineMemberNum() {
        return onlineMemberNum;
    }

    public void setOnlineMemberNum(long onlineMemberNum) {
        this.onlineMemberNum = onlineMemberNum;
    }

    public TIMGroupAddOpt getAddOption() {
        return addOption;
    }

    public void setAddOption(TIMGroupAddOpt addOption) {
        this.addOption = addOption;
    }

    public int getIntAddOption() {
        return intAddOption;
    }

    public void setIntAddOption(int intAddOption) {
        this.intAddOption = intAddOption;
    }

    public boolean isSilenceAll() {
        return isSilenceAll;
    }

    public void setSilenceAll(boolean silenceAll) {
        isSilenceAll = silenceAll;
    }

    public TIMMessage getLastMsg() {
        return lastMsg;
    }

    public void setLastMsg(TIMMessage lastMsg) {
        this.lastMsg = lastMsg;
    }

    public Map<String, byte[]> getCustom() {
        return custom;
    }

    public void setCustom(Map<String, byte[]> custom) {
        this.custom = custom;
    }
    public long getUnreadMessageNum() {
        return unreadMessageNum;
    }

    public void setUnreadMessageNum(long unreadMessageNum) {
        this.unreadMessageNum = unreadMessageNum;
    }
}
